core/common/config/config.go

```diff
@@ -474,6 +478,21 @@ func newP2pConfigWithDefault() P2PConfig {
 	}
 }
 
+func GetHostIp() string {
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return ""
+	}
+	for _, address := range addrs {
+		if ipnet, ok := address.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+			if ipnet.IP.To4() != nil {
```

Comment:
We support IPv6,It is suggested to consider supporting IPv6.

Reply:
Done, now divide GetHostIp() into 2 functions named GetHostIpv4() and GetHostIpv6(), which is be called on different value of the configuration IsIpv6.